src/include/txn.i

```diff
@@ -808,9 +809,13 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, WT
     buf.flags = 0;
 
     /* Check the ondisk value. */
-    if (vpack == NULL)
-        WT_RET(__wt_value_return_buf(cbt, cbt->ref, &buf, &start, &stop));
-    else {
+    if (vpack == NULL) {
+        ret = __wt_value_return_buf(cbt, cbt->ref, &buf, &start, &stop);
+        if (ret != 0) {
+            __wt_buf_free(session, &buf);
+            WT_RET(ret);
```

Comment:
You already checked for `ret != 0`. You can directly call like `return (ret);`

Reply:
@kommiharibabu Yesh I agree, fixed it now.